java/src/main/java/ai/onnxruntime/OnnxTensor.java

```diff
@@ -704,15 +704,20 @@ static OnnxTensor createTensor(
   private static OnnxTensor createTensor(
       OnnxJavaType type, OrtAllocator allocator, Buffer data, long[] shape) throws OrtException {
     int bufferPos;
-    int bufferSize = data.remaining() * type.size;
-    if ((bufferSize < 0) || (bufferSize > ((Integer.MAX_VALUE / type.size) - 10))) {
-      // overflowed as we can't make a direct byte buffer of that size
+    long bufferSizeLong = data.remaining() * (long) type.size;
+    if ((bufferSizeLong < 0) || (bufferSizeLong > (Integer.MAX_VALUE - (8 * type.size)))) {
+      // Negative value implies we overflowed and we can't allocate something that big (though overflowing
+      // a long at this point should be tricky),
+      // and the maximum direct byte buffer size is a little below Integer.MAX_VALUE depending
+      // on the JVM, so we check for something 8 elements below the maximum size.
```

Comment:
Maybe you can try this:
https://wiki.sei.cmu.edu/confluence/display/java/NUM00-J.+Detect+or+prevent+integer+overflow

Reply:
Just for reference, below is how it was done in our C++ code:
https://github.com/microsoft/onnxruntime/blob/rel-1.1.0/include/onnxruntime/core/framework/allocator.h#L226

Reply:
Ah this is a Java thing. You can only allocate a direct byte buffer with fewer than 2^31 elements (and the exact amount below is JVM implementation dependent, but the value I chose should be doable on most compliant JVMs), but I'm doing the multiply using 64-bit ints with values that are guaranteed to be less than 2^31 (because both of the inputs are 32-bit ints before they've been cast to 64-bit ints). It should never overflow the long unless the spec on ByteBuffer changes (and that's a breaking change so is very unlikely to happen). I can remove the negative check and the comment if you want as it's not going to happen.

Reply:
So it won't overflow? I feel remove the negative check would be better, it will make the code simpler.

Reply:
Done.